Backfit cve-2024-28698 to previous version?

Hi All,

Just wondering if this could be backfitted and set out as a small fix to previous versions, as much as I would love to upgrade to CSLA 8.0 I am just not in a position to do this at the moment, but don't want this vulnerability to show up in future testing done by the customer. 

https://www.intruder.io/research/path-traversal-and-code-execution-in-csla-net-cve-2024-28698?utm_content=301411509&amp;utm_medium=social&amp;utm_source=twitter&amp;hss_channel=tw-3189900201

**Version and Platform**
CSLA version: 5.3.3
OS: Window
Platform: WebAPI

Would someone else be able to build it? This is now affecting our pipelines and we are going to have to make them appear as warnings instead of errors which isn't what we really want to do.

##[error]The nuget command failed with exit code(1) and error(NU1904: Warning As Error: Package 'Csla' 5.5.3 has a known critical severity vulnerability, https://github.com/advisories/GHSA-9xhh-3m78-gvgj

--- a/Source/Csla.Shared/Reflection/MethodCaller.cs
+++ b/Source/Csla.Shared/Reflection/MethodCaller.cs
@@ -234,7 +234,11 @@ namespace Csla.Reflection
         string[] splitName = typeName.Split(',');
         if (splitName.Length > 2)
         {
-          var asm = AssemblyLoadContext.Default.LoadFromAssemblyPath(AppContext.BaseDirectory + splitName[1].Trim() + ".dll");
+          var path = AppContext.BaseDirectory + splitName[1].Trim() + ".dll";
+          if (path.Contains("..") || path.Contains(':'))
+            throw new TypeLoadException(path);
+
+          var asm = AssemblyLoadContext.Default.LoadFromAssemblyPath(path);
           return asm.GetType(splitName[0].Trim());
         }
         else

